### PR TITLE
buckets: fix err handling and dir rendering

### PR DESCRIPTION
**Pull request by andrewxhill**

### CODE REVIEW

File: gateway/bucketfs.go [hunk 1]
```diff
@@ -72,8 +72,9 @@ func (f *bucketFileSystem) Exists(bucket, pth string) (bool, string) {
 	ctx, cancel := context.WithTimeout(context.Background(), f.timeout)
 	defer cancel()
 	rep, err := f.client.ListBucketPath(ctx, "", path.Join(bucket, pth), f.auth)
-	if err == nil && !rep.Item.IsDir {
-		return true, ""
+	if err != nil {
+		log.Errorf("error listing bucket path: %v", err)
```
Comment (andrewxhill) [inline gateway/bucketfs.go]: not confident that this log is actually useful... thoughts?

Comment (sanderpick) [reply gateway/bucketfs.go]: Why's that? It's good practice to log an error if you're not returning it.

Comment (sanderpick) [reply gateway/bucketfs.go]: That is to say, looks like a nice addition.

Comment (andrewxhill) [reply gateway/bucketfs.go]: cool, all good then

**Pull request merged**